sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -180,6 +180,70 @@ static inline std::string GetHTTPMessagePreBody(Azure::Core::Http::Request const
 
   return httpRequest;
 }
+
+static void CleanupThread()
+{
+  using namespace Azure::Core::Http::_detail;
+  for (;;)
+  {
+    {
```

Comment:
Why is this nested in an extra layer of curly braces? Remove?

Reply:
nice catch.
On the first design, I was using two mutex and each one was scoped, this extra scope is a fossil of that initial design :)

Reply:
I noticed because such deep nesting is usually a code smell :p
![image](https://user-images.githubusercontent.com/6527137/114957100-265c0a00-9e15-11eb-9f02-3c7b6f3fae90.png)

6 levels of nesting, anything above 3-4 probably should be refactored.